tests/update_workflow_test.go

```diff
@@ -1652,113 +1653,82 @@ func (s *UpdateWorkflowSuite) TestUpdateWorkflow_EmptySpeculativeWorkflowTask_Re
 
 	tv = s.startWorkflow(tv)
 
-	wtHandlerCalls := 0
-	wtHandler := func(task *workflowservice.PollWorkflowTaskQueueResponse) ([]*commandpb.Command, error) {
-		wtHandlerCalls++
-		switch wtHandlerCalls {
-		case 1:
-			// Completes first WT with empty command list.
-			return nil, nil
-		case 2:
-			s.EqualHistory(`
-  1 WorkflowExecutionStarted
-  2 WorkflowTaskScheduled
-  3 WorkflowTaskStarted
-  4 WorkflowTaskCompleted
-  5 WorkflowTaskScheduled // Speculative WT.
-  6 WorkflowTaskStarted
-`, task.History)
-			return nil, nil
-		case 3:
+	// Drain first WFT.
+	_, err := s.PollAndProcessWorkflowTask(tv, taskpoller.DrainTaskFunc)
```

Comment:
:+1:
May be just `DrainWorkflowTask()`? And you don't need comment there any more!

Reply:
are the `()` intentional? I would have left it a variable.